### PR TITLE
feat: metrics explorer compare with other metric part 2

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #12676 

### Description:

CleanShot 2024-12-05 at 19 14 32@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/models/CatalogModel/CatalogModel.ts [hunk 1]
```diff
@@ -852,4 +854,16 @@ export class CatalogModel {
...
+
+    async getAllMetricFields(
+        projectUuid: string,
+        userAttributes: UserAttributeValueMap,
+    ) {
+        return this.database(CatalogTableName)
+            .where('project_uuid', projectUuid)
+            .andWhere('type', CatalogType.Field)
+            .andWhere('field_type', FieldType.METRIC)
+            .andWhere(userAttributesFiltering(userAttributes))
```
Comment (joaoviana) [inline packages/backend/src/models/CatalogModel/CatalogModel.ts:866]: maybe you want tablesConfigurationFiltering as well...?

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts [hunk 2]
```diff
@@ -69,6 +69,8 @@ export class MetricsExplorerService<
         compare: MetricExplorerComparisonType | undefined,
         timeDimensionOverride: TimeDimensionConfig | undefined,
     ): Promise<MetricsExplorerQueryResults> {
+        console.log({ compare, timeDimensionOverride });
```
Comment (joaoviana) [inline packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts:72]: some logs to remove

File: packages/backend/src/models/CatalogModel/CatalogModel.ts (hunk 1 above)
Comment (joaoviana) [reply packages/backend/src/models/CatalogModel/CatalogModel.ts:866]: or just use the `search method...`?

**Comment by joaoviana** [comment]
Some issues with user attribute filtering not working as expected (from the cypress API Tests)

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx [hunk 3]
```diff
@@ -61,6 +65,37 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
...
+
+    const metricsWithTimeDimensionsQuery = useCatalogMetricsWithTimeDimensions({
+        projectUuid,
+        options: {
+            enabled:
+                comparisonType === MetricExplorerComparison.DIFFERENT_METRIC,
+        },
+    });
+
+    const [selectedMetric, setSelectedMetric] =
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:80]: we should reset the `selectedMetric` on close of the modal.

**Comment by joaoviana** [comment]
When I select 'unique payment count' and I select 'event count' as a metric to compare, I get this error: 

```
Error ID: a78c435366c44b759c439097ad4345ef
TypeError: Cannot read properties of undefined (reading 'value')
```

File: packages/backend/src/models/CatalogModel/CatalogModel.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/models/CatalogModel/CatalogModel.ts:866]: can I search without search query? i.e. all metrics?

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts:72]: oh thank you

File: packages/backend/src/services/CatalogService/CatalogService.ts [hunk 4]
```diff
@@ -1075,6 +1090,52 @@ export class CatalogService<
...
+            projectUuid,
+            userAttributes,
+            catalogSearch: {
+                type: CatalogType.Field,
+                filter: CatalogFilter.Metrics,
+            },
+            tablesConfiguration: await this.projectModel.getTablesConfiguration(
+                projectUuid,
+            ),
+        });
```
Comment (joaoviana) [inline packages/backend/src/services/CatalogService/CatalogService.ts:1125]: 👌

**Review by joaoviana — APPROVED** [review]

**Comment by joaoviana** [comment]
> When I select 'unique payment count' and I select 'event count' as a metric to compare, I get this error:
> 
> ```
> Error ID: a78c435366c44b759c439097ad4345ef
> TypeError: Cannot read properties of undefined (reading 'value')
> ```

this has been fixed

**Pull request merged**